io/emoncms/88-emoncms.js

```diff
@@ -80,4 +80,52 @@ module.exports = function(RED) {
         });
     }
     RED.nodes.registerType("emoncms",Emoncms);
+ 
+    function Emoncmsin(n) {
+        RED.nodes.createNode(this,n);
+        this.emonServer = n.emonServer;
+        var sc = RED.nodes.getNode(this.emonServer);
+
+        this.baseurl = sc.server;
+        this.apikey = sc.credentials.apikey;
+
+        this.feedid = n.feedid
+        var node = this;
+        var http;
+        if (this.baseurl.substring(0,5) === "https") { http = require("https"); }
+        else { http = require("http"); }
+        this.on("input", function(msg) {
+            this.url = this.baseurl + '/feed/value.json';
+            this.url += '&apikey='+this.apikey;
+            var feedid = this.feedid || msg.feedid;
+            if (feedid !== "") {
+                this.url += '&id=' + feedid;
+            }
+            if (typeof msg.time !== 'undefined') {
+                this.url += '&time=' + msg.time;
+            }
+            node.log("[emoncms] "+this.url);
+            http.get(this.url, function(res) {
+                node.log("Http response: " + res.statusCode);
+                msg.rc = res.statusCode;
+                msg.payload = "";
+                if ((msg.rc != 200) && (msg.rc != 404)) {
+                    node.send(msg);
+                }
+                res.setEncoding('utf8');
+                res.on('data', function(chunk) {
+                    msg.payload += chunk;
+                });
+                res.on('end', function() {
+                    node.send(msg);
```

Comment:
Does the emoncms api always return JSON here? If so, we ought to parse the body for the user and pass that on.